store/postgres/src/chain_store.rs

```diff
@@ -1028,6 +1028,69 @@ mod data {
                 .collect())
         }
 
+        pub(super) fn clear_call_cache(
+            &self,
+            conn: &PgConnection,
+            from: Option<i32>,
+            to: Option<i32>,
+        ) -> Result<usize, Error> {
+            match self {
+                Storage::Shared => {
+                    use public::eth_call_cache as cache;
+                    let delete_stmt = diesel::delete(cache::table);
+                    match (from, to) {
+                        (Some(from), None) => delete_stmt
+                            .filter(cache::block_number.ge(from))
+                            .execute(conn)
+                            .map_err(Error::from),
+                        (None, Some(to)) => delete_stmt
+                            .filter(cache::block_number.le(to))
+                            .execute(conn)
+                            .map_err(Error::from),
+                        (Some(from), Some(to)) => delete_stmt
+                            .filter(cache::block_number.ge(from))
+                            .filter(cache::block_number.le(to))
+                            .execute(conn)
+                            .map_err(Error::from),
+                        (None, None) => delete_stmt.execute(conn).map_err(Error::from),
+                    }
+                }
+                Storage::Private(Schema { call_cache, .. }) => match (from, to) {
+                    (Some(from), None) => {
+                        let query =
+                            format!("delete from {} where block_number >= $1", call_cache.qname);
+                        sql_query(query)
+                            .bind::<Integer, _>(from)
+                            .execute(conn)
+                            .map_err(Error::from)
+                    }
+                    (None, Some(to)) => {
+                        let query =
+                            format!("delete from {} where block_number <= $1", call_cache.qname);
+                        sql_query(query)
+                            .bind::<Integer, _>(to)
+                            .execute(conn)
+                            .map_err(Error::from)
+                    }
+                    (Some(from), Some(to)) => {
+                        let query = format!(
+                            "delete from {} where block_number >= $1 and block_number <= $2",
+                            call_cache.qname
+                        );
+                        sql_query(query)
+                            .bind::<Integer, _>(from)
+                            .bind::<Integer, _>(to)
+                            .execute(conn)
+                            .map_err(Error::from)
+                    }
+                    (None, None) => {
+                        let query = format!("delete from {}", call_cache.qname);
+                        sql_query(query).execute(conn).map_err(Error::from)
+                    }
```

Comment:
In this case, truncating the table would be faster.
Diesel doesn't have methods for `TRUNCATE` commands, but you can check [how we truncate the block cache](https://github.com/graphprotocol/graph-node/blob/2f9a82d21679614f768a7ac0d3c34587ff168eec/store/postgres/src/chain_store.rs#L1396-L1400) for reference.